hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -345,6 +345,30 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
         "Committed to table {} with the new metadata location {}", fullName, newMetadataLocation);
   }
 
+  @Override
+  public void reset(String metadataFileLocation) {
+    Optional<Long> lockId = Optional.empty();
+    ReentrantLock tableLevelMutex = commitLockCache.get(fullName, t -> new ReentrantLock(true));
+    tableLevelMutex.lock();
+    try {
+      lockId = Optional.of(acquireLock());
+      Table tbl = loadHmsTable();
```

Comment:
I do not like this. Too big part of the commit code is duplicated, and this is especially true for the error handling. This part is complicated and error prone, so we might want to try to find a better way to wave the to parts of the code together.

Reply:
I want to abstract this part of the locking logic, let me think about it~

Reply:
@pvary I abstract the part of the metadata locking logic. Please have a look.